File: 백준/Silver/32396. 차이를 $M$ 이상으로/차이를 $M$ 이상으로.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        long[] nm = Arrays.stream(br.readLine().split(" "))
                .mapToLong(Long::parseLong)
                .toArray();

        long n = nm[0];
        long m = nm[1];

        long[] numbers = Arrays.stream(br.readLine().split(" "))
                .mapToLong(Long::parseLong)
                .toArray();

        int count = 0;
        for (int idx = 1; idx < n; idx++) {
            if (Math.abs(numbers[idx]-numbers[idx-1]) < m) {
                count++;
                idx++;
            }
        }

        System.out.println(count);
    }
}

File: 백준/Silver/32396. 차이를 $M$ 이상으로/차이를 $M$ 이상으로.py
n, m = map(int, input().split())
numbers = list(map(int, input().split()))

count, idx = 0, 1
while idx < n:
    if abs(numbers[idx] - numbers[idx-1]) < m:
        count += 1
        idx += 2
    else:
        idx += 1

print(count)